src/dune_rules/super_context.ml

```diff
@@ -382,7 +382,13 @@ let () =
     let* ctx = Context.DB.by_dir dir in
     let* t = find_exn (Context.name ctx) in
     let* expander = expander t ~dir in
-    Expander.expand_str expander sw |> Action_builder.evaluate_and_collect_facts >>| fst)
+    Expander.expand_str expander sw |> Action_builder.evaluate_and_collect_facts >>| fst);
+  Expander0.set_db (fun ~dir ->
```

Comment:
Are there other places where we potentially should set this db? E.g. `Target.expand_path_from_root`, `Artifacts_db.get_installed_binaries`... (i'm just grepping for usages of `Expander.expand_str`).

Reply:
It's a forward declaration, so it can only be set once. However, there are more cases where we can use this db. For example, in `Artifacts`. I have a PR coming up to do that.